parso/tree.py

```diff
@@ -173,13 +179,116 @@ def get_code(self, include_prefix=True):
             e.g. a statement.
         """
 
+    def search_ancestor(self, *node_types: str) -> 'Optional[BaseNode]':
+        """
+        Recursively looks at the parents of this node or leaf and returns the
+        first found node that matches ``node_types``. Returns ``None`` if no
+        matching node is found.
+
+        :param node_types: type names that are searched for.
+        """
+        node = self.parent
+        while node is not None:
+            if node.type in node_types:
+                return node
+            node = node.parent
+        return None
+
+    def dump(self, *, indent: Optional[Union[int, str]] = None) -> str:
```

Comment:
Shouldn't the default be something like 4? I don't think anybody wants to read a dump with the default `None` :D

But I'm fine with `None` as a default as well. Was just wondering why you picked it.

Reply:
The `None` as default is simply borrowed from [`ast.dump`](https://docs.python.org/3/library/ast.html#ast.dump). Personally I always use `indent=4` for good readability, while `indent=None` is intended for a compact representation.

Reply:
`None` on `ast.dump` was just to preserve backwards compatibility since a lot of people used that function. Since we are just creating this function, it would be cool to have  `4` as the default.

Reply:
Yes, I see the `indent` option of `ast.dump` was just added in Python 3.9, and previously it only generates the single-line compact representation. Let's use `4` for our new function to pretty-print by default.